examples/src/main/java/io/grpc/examples/customloadbalance/ShufflingPickFirstLoadBalancerProvider.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright 2023 The gRPC Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.grpc.examples.customloadbalance;
+
+import io.grpc.LoadBalancer;
+import io.grpc.LoadBalancer.Helper;
+import io.grpc.LoadBalancerProvider;
+import io.grpc.NameResolver.ConfigOrError;
+import io.grpc.Status;
+import java.util.Map;
+
+public class ShufflingPickFirstLoadBalancerProvider extends LoadBalancerProvider {
+
+  private static final String RANDOM_SEED_KEY = "randomSeed";
+
+  @Override
+  public ConfigOrError parseLoadBalancingPolicyConfig(Map<String, ?> rawLoadBalancingPolicyConfig) {
+    Long randomSeed = null;
+
+    // The load balancing configuration generally comes from a remote source over the wire, be
+    // defensive when parsing it.
+    try {
+      if (rawLoadBalancingPolicyConfig.containsKey(RANDOM_SEED_KEY)) {
+        Object randomSeedObj = rawLoadBalancingPolicyConfig.get("randomSeed");
+        if (randomSeedObj instanceof Double) {
+          randomSeed = ((Double) randomSeedObj).longValue();
+        }
+      }
+    } catch (RuntimeException e) {
+      return ConfigOrError.fromError(
+          Status.UNAVAILABLE.withDescription("unable to parse LB config").withCause(e));
+    }
+    return ConfigOrError.fromConfig(new ShufflingPickFirstLoadBalancer.Config(randomSeed));
```

Comment:
Can we move this into the try-catch. The constructor could have a null-pointer check or similar.

Reply:
Done. Did not add a null check - it is ok for the config to have a null value for the random seed.

Reply:
I wasn't suggesting to make a null check. Just that "calling the constructor might also throw" (if not now, maybe in the future) and so we want to include it in the try-catch.